### ISSUE TITLE
NullPointerException when setting mediaItem

### ISSUE THREAD

**Issue by XilinJia** [androidx/media#1333]
### Version

Media3 1.3.1

### Devices that reproduce the issue

S21 Android 14

### Reproducible in the demo app?

Not tested

### Reproduction steps

```
val builder = MediaMetadata.Builder()
        builder.setArtist(p.getFeedTitle())
        builder.setTitle(p.getEpisodeTitle())
        builder.setAlbumArtist(p.getFeedTitle())
        builder.setDisplayTitle(p.getEpisodeTitle())
        builder.setSubtitle(p.getFeedTitle())
val mediaItem = MediaItem.Builder().setMediaMetadata(builder.build()).build()
exoPlayer?.setMediaItem(mediaItem)
```

### Expected result

expect the mediaitem being set

### Actual result

```
024-05-01 11:48:33.516  3770-3770  System.err              ac.mdiq.podcini.debug                W  java.lang.NullPointerException
2024-05-01 11:48:33.516  3770-3770  System.err              ac.mdiq.podcini.debug                W  	at androidx.media3.common.util.Assertions.checkNotNull(Assertions.java:155)
2024-05-01 11:48:33.516  3770-3770  System.err              ac.mdiq.podcini.debug                W  	at androidx.media3.exoplayer.source.DefaultMediaSourceFactory.createMediaSource(DefaultMediaSourceFactory.java:460)
2024-05-01 11:48:33.516  3770-3770  System.err              ac.mdiq.podcini.debug                W  	at androidx.media3.exoplayer.ExoPlayerImpl.createMediaSources(ExoPlayerImpl.java:1931)
2024-05-01 11:48:33.517  3770-3770  System.err              ac.mdiq.podcini.debug                W  	at androidx.media3.exoplayer.ExoPlayerImpl.setMediaItems(ExoPlayerImpl.java:586)
2024-05-01 11:48:33.517  3770-3770  System.err              ac.mdiq.podcini.debug                W  	at androidx.media3.common.BasePlayer.setMediaItems(BasePlayer.java:56)
2024-05-01 11:48:33.517  3770-3770  System.err              ac.mdiq.podcini.debug                W  	at androidx.media3.common.BasePlayer.setMediaItem(BasePlayer.java:41)
2024-05-01 11:48:33.517  3770-3770  System.err              ac.mdiq.podcini.debug                W  	at ac.mdiq.podcini.playback.service.PlaybackService.updateMediaSessionMetadata(PlaybackService.kt:1171)
```

### Media

not related to any specific media

### Bug Report

- [ ] You will email the zip file produced by `adb bugreport` to android-media-github@google.com after filing this issue.

**Comment by MGaetan89** [comment]
I think that you forgot to set the media uri on your `MediaItem`, ie.:
```kotlin
val mediaItem = MediaItem.Builder()
    .setUri("https://my.media/".toUri())
    .setMediaMetadata(builder.build())
    .build()
```

**Comment by XilinJia** [comment]
> I think that you forgot to set the media uri on your `MediaItem`, ie.:
> 
> ```kotlin
> val mediaItem = MediaItem.Builder()
>     .setUri("https://my.media/".toUri())
>     .setMediaMetadata(builder.build())
>     .build()
> ```
Merci beaucoup @MGaetan89. That cut it.  

Hope I somewhere in the doc state that setUri is necessary.

**Comment by tonihei** [comment]
Yes, you need a `Uri` to play the item with `ExoPlayer`.

I'm curious what you expected to happen without the Uri. Could you explain that in more detail? The documentation [usually shows examples with `Uri`](https://developer.android.com/media/media3/exoplayer/hello-world#populate-playlist), but it doesn't point out that it is definitely needed assuming there is no way you can play this item without a `Uri`.

**Comment by XilinJia** [comment]
I migrated the project from exoplayer2 to media3.  In the way, it's more convenient to do: `exoPlayer?.setMediaSource(mediaSource!!, false)` and then play the media.  That works without setting mediaItem.  But it appears (?) that I need to set the mediaItem in order to show the media info on the notification widget, so I started doing the above and got the exception.

So now since exoplayer requires setting of the uri, the separate setting of media source is redundant, or even overridden?  What's the best approach?
